ion-hash/src/element_hasher.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright Amazon.com, Inc. or its affiliates.
+
+//! Provides [`ElementHasher`], a single-use IonHash implementation for an
+//! [`Element`].
+
+use std::io;
+
+use digest::{FixedOutput, Output, Reset, Update};
+use ion_rs::result::IonResult;
+use ion_rs::value::Element;
+
+use crate::representation::RepresentationEncoder;
+use crate::type_qualifier::TypeQualifier;
+use crate::Markers;
+
+pub(crate) struct ElementHasher<D>
+where
+    D: Update + FixedOutput + Reset + Clone + Default,
+{
+    pub(crate) digest: D,
+}
+
+impl<D> ElementHasher<D>
+where
+    D: Update + FixedOutput + Reset + Clone + Default,
+{
+    pub(crate) fn new(digest: D) -> ElementHasher<D> {
+        ElementHasher { digest }
+    }
+
+    pub(crate) fn hash_element<E>(mut self, elem: &E) -> IonResult<Output<D>>
+    where
+        E: Element + ?Sized,
+    {
+        self.update_serialized_bytes(elem)?;
+        Ok(self.digest.finalize_fixed())
+    }
```

Comment:
To line up our traits, I think this could be  `impl IonHasher for ElementHasher`, though this is a very minor factoring change as this is an internal wrapper.

Reply:
Whoops, yeah.